### PR TITLE
Ensure we remove translation of source that have been removed

### PR DESCRIPTION
**Pull request by mardiros**
Hello @kaedroho 

We have currently a bug in the translation.

The flow is:

 we have a field named "strapline", and we have to clean it in some pages.

 * in the translation editor, the source disapear

 * after publishing, the translation is still there because it has not been cleaned in the translated page.

All tables (wagtail_localized_string, wagtail_localized_stringtranslation, ...) are cleaned but the data is still in the published page.

Here is my pull request for the fix.

Tell me if you need more explanation of the bug.

### CODE REVIEW

File: wagtail_localize/segments/ingest.py [hunk 1]
```diff
@@ -226,10 +227,23 @@ def ingest_segments(original_obj, translated_obj, src_locale, tgt_locale, segmen
...
-        if hasattr(field, "restore_translated_segments"):
+        if not field_segments:
+
+            if isinstance(field, (models.TextField, models.CharField)):
+                data = field.value_from_object(original_obj)
+            else:
+                # TODO implement for other types
+                continue
+
+            setattr(translated_obj, field_name, data)
```
Comment (kaedroho) [inline wagtail_localize/segments/ingest.py:244]: To synchronise a field with the source, you should register it as a ``SynchronisedField``. I don't think localize should try to be clever about fields that are not configured

File: wagtail_localize/segments/ingest.py [hunk 2]
```diff
@@ -226,10 +227,23 @@ def ingest_segments(original_obj, translated_obj, src_locale, tgt_locale, segmen
         field_name, segment = segment.unwrap()
         segments_by_field_name[field_name].append(segment)
 
-    for field_name, field_segments in segments_by_field_name.items():
+    for field in get_translatable_fields(translated_obj.__class__):
```
Comment (kaedroho) [inline wagtail_localize/segments/ingest.py:230]: Makes sense, can't remember why I didn't do this before!

File: wagtail_localize/segments/ingest.py [hunk 3]
```diff
@@ -226,10 +227,23 @@ def ingest_segments(original_obj, translated_obj, src_locale, tgt_locale, segmen
...
+
+        field_name = field.field_name
+        field_segments = segments_by_field_name[field_name]
         field = translated_obj.__class__._meta.get_field(field_name)
 
-        if hasattr(field, "restore_translated_segments"):
+        if not field_segments:
+
+            if isinstance(field, (models.TextField, models.CharField)):
```
Comment (kaedroho) [inline wagtail_localize/segments/ingest.py:238]: Maybe we should synchronise all translatable fields before restoring translations? We already do this for streamfield

**Comment by zerolab** [comment]
@mardiros I believe this is now handled by #622 / #635 and included in v1.3.x
I realise it took some time, but would appreciate a check on your side

**Comment by mardiros** [comment]
Thanks @zerolab 

I close this pull request, and will open an issue if something will go wrong on my next update.

**Pull request closed without merging**